### PR TITLE
Fix pie rows settings include filtered out values

### LINKED ISSUE #47867: slices removed from query result still appear in pie chart settings
### Describe the bug

https://github.com/user-attachments/assets/0eeaa36b-1d73-4d06-b8bf-abc54c89d285

After the query result changes, the rows that are no longer in the query result should not have slices visible in the viz settings UI.

### To Reproduce

1. New question -> Count of orders by created at: month, limit 5
2. Render as pie chart, move some rows around
3. Go back to notebook editor, change row limit to 2
4. Go back to viz settings sidebar, all 5 slices still visible

### Expected behavior

Only the current rows from the query result should have visible slices in the settings UI

### Logs

_No response_

### Information about your Metabase installation

```JSON
stats, commit hash 0064a6a
```

### Severity

Can make the pie chart difficult to use, makes the currently in development sunburst chart incorrect when switching dimension columns

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/47867

### Description

After editing pie rows settings, if dataset returned fewer rows we still showed previously existing rows in the settings sidebar. This PR filters them out.

### How to verify

- Create a Pie chart, reorder rows
- Apply filter or row limit to remove some rows from the dataset
- Ensure the settings sidebar shows rows that only exist in the dataset

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by EmmadUsmani — COMMENTED** [review]
The problem with this approach is that when you edit `pie.rows` and save the question after rows have been removed from the query result, when those rows reappear they will no longer have their settings. This is why we kept track of the removed rows and marked them with `hidden: true` instead of removing them from the setting. See the example below

https://github.com/user-attachments/assets/ce3f293e-4052-4032-928c-efe0e2913248

**Comment by alxnddr** [comment]
@EmmadUsmani thanks, I fixed that

File: frontend/src/metabase/visualizations/shared/settings/pie.ts [hunk 1]
```diff
@@ -359,4 +361,10 @@ export function getPieRows(
   return newPieRows;
 }
 
+export const getPieSortRowsDimensionSetting = (
+  settings: ComputedVisualizationSettings,
+) => {
+  return settings["pie.dimension"];
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/shared/settings/pie.ts:367]: Is this supposed to be the same as `pie.dimension`?

**Review by EmmadUsmani — APPROVED** [review]
Looks good 👍

**Pull request merged**